### ISSUE TITLE
concurrent map read/write fatal error after a few hours of operation

### ISSUE THREAD

**Issue by ehthayer** [linkedin/Burrow#257]
Runs for a few hours and gets a concurrent map read/write error.
[burrow.log](https://github.com/linkedin/Burrow/files/1431376/burrow.log)

**Comment by daodennis** [comment]
@ehthayer this looks like the same thing as https://github.com/linkedin/Burrow/issues/169 there are technically two separate traces in that bug, so could be there are two issues. I'm building off of master fwiw.

**Comment by ehthayer** [comment]
```created by main.NewOffsetStorage.func1
	/home/ethayer/go/src/github.com/linkedin/Burrow/offsets_store.go:140 +0x35e
``` looked suspicious in the trace I saw.

**Comment by toddpalino** [comment]
Yeah, there's definitely some concurrency issues in a couple places. It's being fixed soon.

**Comment by toddpalino** [comment]
Resolved by 1.0 rewrite
